eclipse-theia/theia
# feat: rag based variable provider

#### What it does

Adds a RAG based variable provider specialized for the GLSP documentation.

It uses `langchain` for performing RAG

#### How to test

- clone the `git@github.com:eclipse-glsp/glsp-website-source.git` repository
- copy this changelog `https://github.com/eclipse-glsp/glsp-server-node/blob/main/CHANGELOG.md` to the `content/documentation` root directory

```bash
git clone git@github.com:eclipse-glsp/glsp-website-source.git
curl https://github.com/eclipse-glsp/glsp-server-node/raw/main/CHANGELOG.md -o glsp-website-source/content/documentation/CHANGELOG.md
```

Either modify the `GLSPVariableContribution` to default to your clone of the source repo or hand it over as a parameter in the chat.

In the Chat View, ask GLSP questions with and without `#glsp` and compare the difference. See the log for the RAG output.

#### Follow-ups



#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
